### DISCUSSION TITLE
Firmware and Analog input

### DISCUSSION THREAD

**Discussion by DominusFulguris** [FundamentalFrequency/.github#39] [MCAD/ECAD]
I am confused by this comment in the firmware source:
"Using a filtered analog kind of worked, but for some reason it would update whenever any other button was pressed as well"

Stone, could you explain what the issue is?

Also, I think I am seeing that only one axis of the joystick is actually being used. Is that intentional?

**Comment by stonepreston** [comment]
I had a lot of issues using the pitch bend potentiometer class and the filtered analog class in control surface with the teensy 4.1, They work fine with the teensy 2.0, but seem to have issues with the faster clock speed maybe on the 4.1. Id prefer to use the pitch bend potentiometer class, but it was super noisy for some reason. I then tried to use the filtered analog that pitch bend potentiometer uses internally (which should in theory filter out the noise), but it would update whenever buttons were pressed (ie it would send a pitch bend midi message any time any button was pressed, instead of only when the pitch bend pot was changed. [Here](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d4e/Pitch-Bend-Potentiometer_8ino-example.html) is an example of using the pitch bend pot. That was super easy to use on the 2.0, but does not seem to work well on the 4.1

Yes currently only one axis is used as I had nothing else to map really. Its possible we could use the other axis it for something else in the future

**Reply by DominusFulguris** [reply]
"but it would update whenever buttons were pressed (ie it would send a pitch bend midi message any time any button was pressed, instead of only when the pitch bend pot was changed"

I'm not sure I understand why that behaviour is undesirable. Sure it is sending extra data but at the same time it ensures that any interpretation of a button press includes current pitch bend state that might affect it, avoiding the need of the upstream device from having to retain that state.

**Reply by DominusFulguris** [reply]
As to noise, the current PCB could be improved. You have a high-impedance source (joystick pot wiper) feeding into a very high impedance sink (ADC pin) with pretty long traces in between.  This is a pretty good way to pick up noise.  Adding a filter cap to the ADC pin near the Teensy would help.

**Reply by stonepreston** [reply]
I did try using the same joystick with a cap in a breadboard setup to try and reduce noise, but it did not seem to help in my initial tests. The normal behavior of the pitch bend potentiometer object is to only send data when the value changes, although you are correct that its not a huge deal in the end IF the noise was not present

I opened an issue about it in control surface [here](https://github.com/tttapa/Control-Surface/issues/726). Although he was unable to reproduce.
